pyproject.toml

```diff
@@ -0,0 +1,43 @@
+[build-system]
+requires = ["setuptools", "versioningit"]
+build-backend = "setuptools.build_meta"
+
+[project]
+name = "chirp"
+dynamic = ["version"]
+description = "A cross-platform cross-radio programming tool"
+requires-python = ">=3.7,<4"
+dependencies = [
+    "pyserial",
+    "requests",
+    "six",
+    "future",
+    "importlib-resources;python_version<'3.10'",
+    "yattag",
+    "importlib-metadata;python_version<'3.8'"
+]
+
+[tool.setuptools]
+include-package-data = true
+
+[tool.setuptools.packages.find]
+include = ["chirp*"]
+
+[project.urls]
+homepage = "https://chirp.danplanet.com"
+repository = "https://github.com/kk7ds/chirp"
+
+[project.optional-dependencies]
+wx = ["wxPython"]
+
+[project.scripts]
+chirp = "chirp.wxui:chirpmain"
+chirpc = "chirp.cli.main:main"
+
+[tool.versioningit.tag2version]
+rmprefix = "release_"
+
+[tool.versioningit.format]
+distance = "{committer_date:%Y%m%d}"
+dirty = "{committer_date:%Y%m%d}+d{build_date:%Y%m%d}"
+distance-dirty = "{committer_date:%Y%m%d}+d{build_date:%Y%m%d}"
```

Comment:
Ah, I didn't pay attention here. I don't think I want to do this. I want to know when people are running from git versus one of the specific builds. Someone running in the middle of two commits that went out on a specific day will look like they're running the same thing as someone on a formal build but they're not. Also, some of our agreements with network services rely on our ability to enumerate builds within a certain window for anti-spam and anti-leech abilities. This will basically make it so that every day becomes a build number even though that's not true.

As mentioned, there's more to the versioning that we use internally than just display and it really matters to me (and others) that we're able to be specific.

Reply:
This only works with a fully-cloned repository, ie full history. Otherwise this errors out since `committer_date` relies on a full history metadata to retrieve the commit date. When using an archive, shallow clone, etc, whatever is in `PKG-INFO` overrides any computed version.

Can also add the git hash in there to further delineate something built directly from git however.

Reply:
So we could make this just the githash then? or maybe `git-$githash`? That'd make me happy.

Reply:
Hash is far more useful.

Reply:
`{revision}` is the hash.